### PR TITLE
add domain in JS advisories

### PR DESCRIPTION
**Pull request by ripienaar**
Signed-off-by: R.I.Pienaar <rip@devco.net>

/cc @nats-io/core

### CODE REVIEW

**Comment by ripienaar** [comment]
Tests don't deadlock but a 2nd pair of eyes over locks would be good

File: server/consumer.go [hunk 1]
```diff
@@ -836,6 +836,7 @@ func (o *consumer) sendDeleteAdvisoryLocked() {
 		Stream:   o.stream,
 		Consumer: o.name,
 		Action:   DeleteEvent,
+		Domain:   o.js.config.Domain,
```
Comment (derekcollison) [inline server/consumer.go:839]: I would pull from server opts. All assets have srv property.

So opts := o.srv.getOpts()
opts.JetStreamDomain.

File: server/jetstream.go [hunk 2]
```diff
@@ -341,6 +341,7 @@ func (s *Server) handleOutOfSpace(stream string) {
 			ServerID: s.ID(),
 			Stream:   stream,
 			Cluster:  s.cachedClusterName(),
+			Domain:   s.js.config.StoreDir,
```
Comment (derekcollison) [inline server/jetstream.go:344]: Same comment as above.

File: server/jetstream_api.go [hunk 3]
```diff
@@ -828,6 +828,30 @@ func (s *Server) getRequestInfo(c *client, raw []byte) (pci *ClientInfo, acc *Ac
 	return &ci, acc, hdr, msg, nil
 }
 
+func (a *Account) jsDomain() string {
```
Comment (derekcollison) [inline server/jetstream_api.go:831]: Can remove, or simplify as above. Accounts also have srv.

File: server/jetstream_api.go [hunk 4]
```diff
@@ -2638,7 +2662,7 @@ func (s *Server) processStreamRestore(ci *ClientInfo, acc *Account, cfg *StreamC
 	s.Noticef("Starting restore for stream '%s > %s'", acc.Name, streamName)
 
 	start := time.Now().UTC()
-
+	domain := js.config.Domain
```
Comment (derekcollison) [inline server/jetstream_api.go:2665]: Same here with s.getOpts()

File: server/jetstream_cluster.go [hunk 5]
```diff
@@ -3086,6 +3090,7 @@ func (s *Server) sendConsumerLeaderElectAdvisory(o *consumer) {
 		Consumer: consumer,
 		Leader:   s.serverNameForNode(node.GroupLeader()),
 		Replicas: s.replicas(node),
+		Domain:   acc.jsDomain(),
```
Comment (derekcollison) [inline server/jetstream_cluster.go:3093]: just do s.jsDomain() imo.

**Review by derekcollison — CHANGES_REQUESTED** [review]

**Pull request merged**